tests/Compat/FormDecorator/LabelDecoratorTest.php

```diff
@@ -23,4 +27,52 @@ public function testWithoutLabelAttribute(): void
 
         $this->assertSame('&nbsp;', $results->assemble()->render());
     }
+
+    public function testWithRequiredField(): void
+    {
+        $formElement = new TextElement('test', [
+            'required' => true,
+            'label' => 'test-label'
+        ]);
+
+        $results = new FormElementDecorationResult();
+        $form = (new CompatForm())->addElement($formElement);
```

Comment:
Unittests should be lightweight and not depend on unrelated components or implementation details. You are instantiating a form only, because `decorateForm` requires one. Instead, please create and use a stub or mock.